src/Symfony/Component/Console/Helper/TreeBuilder.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class TreeBuilder
+{
+    public static function fromArray(array $array, ?string $root = null): TreeNode
```

Comment:
```suggestion
    /**
     * Parse an array to build a TreeNode.
     *
     * Usage:
     * <code>
     *   TreeBuilder::fromArray([
     *      'rootKey' => [
     *          'parent' => [
     *               'children => [
     *                   'child1',
     *                   'child2',
     *                ],
     *           ],
     *         'orphan' => 'orphan'
     *    ]
     * ]);
     * </code>
     */
    public static function fromArray(array $array, ?string $root = null): TreeNode
```

When I tried it on a concrete example, I take a moment to look for "how" to map things, maybe provided an example could help a little bit :)

Reply:
Doc will be made for this, yes. But the idea was not to need 'orphan' value here

```php
[
    'a' => [
        'a1,
        'a2'
    ],
    'b',
    'c' => [
        'c1',
        'c2' => [
            'c21' => [
                'c211',
                'c212,
            ],
        ],
    ],
]
```